anesthetic/plot.py

```diff
@@ -323,11 +323,28 @@ def kde_plot_1d(ax, data, *args, **kwargs):
     xmax = kwargs.pop('xmax', None)
     weights = kwargs.pop('weights', None)
     ncompress = kwargs.pop('ncompress', 1000)
+    q = kwargs.pop('q', '5sigma')
+    if isinstance(q, str):
+        sigmas = {'1sigma': 0.682689492137086,
+                  '2sigma': 0.954499736103642,
+                  '3sigma': 0.997300203936740,
+                  '4sigma': 0.999936657516334,
+                  '5sigma': 0.999999426696856}
+        q = (1 - sigmas[q]) / 2
+    if isinstance(q, float) or isinstance(q, int):
+        if q > 0.5:
+            q = 1 - q
+        q = (q, 1-q)
+
+    if weights is not None:
+        data = data[weights != 0]
+        weights = weights[weights != 0]
+
     x, w = sample_compression_1d(data, weights, ncompress)
     kde = gaussian_kde(x, weights=w)
     p = kde(x)
     p /= p.max()
-    i = ((x < quantile(x, 0.999, w)) & (x > quantile(x, 0.001, w))) | (p > 0.1)
+    i = ((x > quantile(x, q[0], w)) & (x < quantile(x, q[1], w)))
```

Comment:
Should the same thing be done for `fastkde`?

Reply:
Quantile-wise yes, although the compression does not apply to fastkde.

Are you currently adding tests, or should I review this in toto?

Reply:
writing tests